### PR TITLE
master

### PR DESCRIPTION
**Pull request by aonofriichuk**

### CODE REVIEW

File: pqueue/pqueue.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ type PQueue struct {
 	database *db.DataStorage
 	// Queue settings.
 	settings *PQueueSettings
+
+	newMsg chan bool
```
Comment (vburenin) [inline pqueue/pqueue.go]: newMsgNotification?

File: pqueue/pqueue.go [hunk 2]
```diff
@@ -68,6 +70,7 @@ func initPQueue(database *db.DataStorage, queueName string, settings *PQueueSett
 		ACTION_SET_LOCK_TIMEOUT:    pq.SetLockTimeout,
 		ACTION_UNLOCK_BY_ID:        pq.UnlockMessageById,
 	}
+	pq.newMsg = make(chan bool)
```
Comment (vburenin) [inline pqueue/pqueue.go]: move it into base struct init pls.

File: pqueue/pqueue.go [hunk 3]
```diff
@@ -134,6 +137,10 @@ func (pq *PQueue) storeMessage(msg *PQMessage, payload string) error {
 	pq.allMessagesMap[msg.Id] = msg
 	pq.trackExpiration(msg)
 	pq.availableMsgs.Push(msg.Id, msg.Priority)
+	select {
```
Comment (vburenin) [inline pqueue/pqueue.go]: Is it necessary to push message each time if queue is not empty?

File: pqueue/pqueue.go [hunk 4]
```diff
@@ -169,6 +176,43 @@ func (pq *PQueue) PopMessage() (queue_facade.IMessage, error) {
...
+	msg := make([]queue_facade.IMessage, 0, limit)
+	// Run timeout control routine
+	timeout := make(chan bool, 1)
+	go func() {
+		time.Sleep(time.Duration(timeoutMsec) * time.Millisecond)
+		timeout <- true
+	} ()
+	// Pop messages from the queue
+	needExit := false
+	for int64(len(msg)) < limit && !needExit{
```
Comment (vburenin) [inline pqueue/pqueue.go]: no need to cast a type, just make the same type for the interface.

File: pqueue/pqueue.go (hunk 3 above)
Comment (aonofriichuk) [inline pqueue/pqueue.go]: Ok, will send the message only if msg was added in empty queue. PopWait will handle this usecase.

**Pull request merged**